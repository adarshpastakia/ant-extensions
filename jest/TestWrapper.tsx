// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import i18next from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18next
  // pass the i18n instance to react-i18next.
  .use(initReactI18next)
  // init i18next
  // for all options read: https://www.i18next.com/overview/configuration-options
  .init({
    resources: {
      en: {}
    },
    defaultNS: "core",
    fallbackLng: ["en"],
    keySeparator: ".",

    interpolation: {
      escapeValue: false // not needed for react as it escapes by default
    }
  });
i18next.languages = ["en", "ar"];

export const TestWrapper: React.FC = ({ children }) => {
  return <I18nextProvider i18n={i18next}>{children}</I18nextProvider>;
};
