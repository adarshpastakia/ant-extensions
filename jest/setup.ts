// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import "@testing-library/jest-dom";
import "jsdom-global";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  }))
});
