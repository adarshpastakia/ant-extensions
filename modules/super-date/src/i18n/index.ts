// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import i18next from "i18next";
import en from "./en.json";
import ar from "./ar.json";
import { I18nKey } from "../utils/i18nKey";

const callback = () => {
  i18next.addResourceBundle("en", I18nKey, en, true, false);
  i18next.addResourceBundle("ar", I18nKey, ar, true, false);
};

if (i18next.isInitialized) {
  callback();
} else {
  i18next.on("initialized", callback);
}
