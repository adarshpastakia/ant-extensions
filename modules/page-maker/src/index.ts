// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import "./i18n";

export { EnumTypes } from "./utils/types";
export type { PageConfig, PageItem, WidgetObject } from "./utils/types";
export { PageMaker } from "./components/PageMaker";
