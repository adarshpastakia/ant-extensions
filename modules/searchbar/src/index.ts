// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT
import "./i18n";

export {Type, Operator} from "./utils/types";
export type { IFilterObject, IFilterField } from "./utils/types";
export { SearchBar } from "./components/SearchBar";
