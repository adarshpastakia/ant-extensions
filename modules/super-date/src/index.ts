// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT
import "./i18n";

export { InlineDatePicker } from "./components/InlineDatePicker";
export { InlineRangePicker } from "./components/InlineRangePicker";
export { RelativeDatePicker } from "./components/RelativeDatePicker";
export { RelativeRangePicker } from "./components/RelativeRangePicker";
export { DateUtils } from "./utils/dateUtils";
export { DateParts } from "./utils/types";
