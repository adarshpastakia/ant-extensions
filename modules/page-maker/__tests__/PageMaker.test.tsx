// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import { PieChartOutlined } from "@ant-design/icons";
import { render, RenderResult } from "@testing-library/react";
import React from "react";
import { TestWrapper } from "../../../jest/TestWrapper";
import { EnumTypes, PageConfig, PageMaker, WidgetObject } from "../src";

describe("PageMaker", () => {
  let fragment: RenderResult;

  const widgets: WidgetObject[] = [
    {
      id: "widget-1",
      icon: <PieChartOutlined />,
      title: "Test Widget"
    }
  ];

  const config: PageConfig = [
    {
      id: "head-1",
      type: EnumTypes.HEADING,
      title: "Heading",
      color: "#487eb0",
      size: 18
    },
    {
      id: "div-1",
      type: EnumTypes.DIVIDER
    },
    {
      id: "row-2",
      type: EnumTypes.ROW,
      height: 400,
      children: [
        {
          id: "col-2",
          type: EnumTypes.COL,
          colSpan: 6,
          children: [
            {
              id: "tile-2",
              widgetId: widgets[0].id,
              type: EnumTypes.TILE,
              title: "Tile head",
              bordered: true,
              expandable: true,
              color: "#227093",
              info:
                "<b>Information Card</b>\nThis tooltip can display information for the <u>widget</u>\nCan include <em>HTML tags</em>",
              iconCls: "mdi mdi-chart-pie"
            }
          ]
        },
        {
          id: "col-3",
          type: EnumTypes.COL,
          colSpan: 6,
          children: []
        }
      ]
    }
  ];

  beforeEach(() => {
    fragment = render(
      <PageMaker config={config} widgets={widgets} renderWidget={(widgetId) => <div />} />,
      {
        wrapper: TestWrapper
      }
    );
  });

  afterAll(() => {
    fragment.unmount();
  });

  it("should render", (done) => {
    expect(fragment.container).toMatchSnapshot();
    done();
  });

  it.todo("should fire change event");
});
