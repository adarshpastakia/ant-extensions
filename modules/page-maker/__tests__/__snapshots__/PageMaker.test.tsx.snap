// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMaker should render 1`] = `
<div>
  <div
    class="ant-ext-pm__container"
  >
    <div
      class="ant-ext-pm__page"
      data-editing="false"
      data-type="page"
    >
      <div
        class="ant-ext-pm__grid"
      >
        <div
          class="ant-ext-pm__item"
          data-expanded="false"
          data-hover="false"
          data-id="head-1"
          data-type="heading"
        >
          <div
            style="color: rgb(72, 126, 176); font-size: 18px;"
          >
            <div>
              Heading
            </div>
          </div>
        </div>
        <div
          class="ant-ext-pm__item"
          data-expanded="false"
          data-hover="false"
          data-id="div-1"
          data-type="divider"
        >
          <div
            class="ant-divider ant-divider-horizontal"
            role="separator"
            style="font-size: 13px;"
          />
        </div>
        <div
          class="ant-ext-pm__item"
          data-expanded="false"
          data-hover="false"
          data-id="row-2"
          data-type="row"
          style="height: 400px;"
        >
          <div
            class="ant-ext-pm__rowContainer"
          >
            <div
              class="ant-ext-pm__item"
              data-expanded="false"
              data-hover="false"
              data-id="col-2"
              data-type="col"
              style="grid-column-end: span 6; grid-auto-rows: auto;"
            >
              <div
                class="ant-ext-pm__item"
                data-expanded="false"
                data-hover="false"
                data-id="tile-2"
                data-type="tile"
              >
                <div
                  class="ant-ext-pm__tileHead"
                >
                  <span
                    style="color: rgb(34, 112, 147);"
                  >
                    <i
                      class="mdi mdi-chart-pie"
                    />
                  </span>
                  <label
                    style="color: rgb(34, 112, 147);"
                  >
                    Tile head
                  </label>
                  <div>
                    <button>
                      <span
                        aria-label="info-circle"
                        class="anticon anticon-info-circle"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="info-circle"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                          />
                          <path
                            d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                          />
                        </svg>
                      </span>
                    </button>
                    <button
                      class="ant-ext-pm__tileExpander"
                    >
                      <span
                        aria-label="fullscreen"
                        class="anticon anticon-fullscreen"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="fullscreen"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M290 236.4l43.9-43.9a8.01 8.01 0 00-4.7-13.6L169 160c-5.1-.6-9.5 3.7-8.9 8.9L179 329.1c.8 6.6 8.9 9.4 13.6 4.7l43.7-43.7L370 423.7c3.1 3.1 8.2 3.1 11.3 0l42.4-42.3c3.1-3.1 3.1-8.2 0-11.3L290 236.4zm352.7 187.3c3.1 3.1 8.2 3.1 11.3 0l133.7-133.6 43.7 43.7a8.01 8.01 0 0013.6-4.7L863.9 169c.6-5.1-3.7-9.5-8.9-8.9L694.8 179c-6.6.8-9.4 8.9-4.7 13.6l43.9 43.9L600.3 370a8.03 8.03 0 000 11.3l42.4 42.4zM845 694.9c-.8-6.6-8.9-9.4-13.6-4.7l-43.7 43.7L654 600.3a8.03 8.03 0 00-11.3 0l-42.4 42.3a8.03 8.03 0 000 11.3L734 787.6l-43.9 43.9a8.01 8.01 0 004.7 13.6L855 864c5.1.6 9.5-3.7 8.9-8.9L845 694.9zm-463.7-94.6a8.03 8.03 0 00-11.3 0L236.3 733.9l-43.7-43.7a8.01 8.01 0 00-13.6 4.7L160.1 855c-.6 5.1 3.7 9.5 8.9 8.9L329.2 845c6.6-.8 9.4-8.9 4.7-13.6L290 787.6 423.7 654c3.1-3.1 3.1-8.2 0-11.3l-42.4-42.4z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="ant-ext-pm__tileBody"
                >
                  <div />
                </div>
              </div>
            </div>
            <div
              class="ant-ext-pm__item"
              data-expanded="false"
              data-hover="false"
              data-id="col-3"
              data-type="col"
              style="grid-column-end: span 6; grid-auto-rows: max-content;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
